### PR TITLE
Biosys 160

### PR DESCRIPTION
**Pull request by chungvl**
Added about, dataset storage via StorageService, settings stub, about stub

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -38,6 +40,17 @@ export class AppComponent implements OnInit {
             } else {
                 this.nav.setRoot('HomePage');
             }
+            
+            /*  FIXME: The backend needs to support the dataset API endpoint WITHOUT authentication,
+                or some sort of weird default auth_token needs to be created, so as to avoid stopping
+                a user from entering data despite not having logged in. */
+            localStorage.setItem('auth_token', '27eb3ec6b0370f78b7bd5c3ad865c30cf7ea3493');
+            this.api.getDatasets().subscribe(datasets =>{
+                for (let i=0; i < datasets.length; i++) {
```
Comment (tony-gaia) [inline src/app/app.component.ts]: FYI you can also do:

```
for (let dataset of datasets) {
     this.store.putDataset(dataset);
}
```

File: src/app/app.module.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ import { ObservationPageModule } from "../pages/observation/observation.module";
         IonicModule.forRoot(AppComponent),
         IonicStorageModule.forRoot(),
         ComponentsModule,
-        ObservationPageModule
+        ObservationPageModule // FIXME: FAB buttons don't work without this
```
Comment (tony-gaia) [inline src/app/app.module.ts]: just put @IonicPage() above the component declaration of the ObservationPage @Component

File: src/pages/home/home.ts [hunk 3]
```diff
@@ -38,17 +38,20 @@ class FooBoo implements Record {
...
     }
 
-    public clickedSync() {
+    public clickedUpload() {
         return;
     }
     
-    public clickedNew(type: DataType) {
-        this.navCtrl.setRoot(ObservationPage);
+    public clickedNew(type: DataType=undefined, fabContainer: FabContainer) {
```
Comment (tony-gaia) [inline src/pages/home/home.ts]: No reason to ever give 'undefined' as a default value (this is the default). If you want to make this an optional argument, do like:

```typescript
public onClickedNew(datatype?: Datatype)
```
Secondly, if you do use optional arguments, they must always go after the mandatory ones.

Third, why do you need to pass in fabContainer if you've made it an instance variable (i.e. fabNew) of the class?

File: src/pages/home/home.ts [hunk 4]
```diff
@@ -38,17 +38,20 @@ class FooBoo implements Record {
...
+        fabContainer.close();
+        this.navCtrl.push(ObservationPage);
         return;
```
Comment (tony-gaia) [inline src/pages/home/home.ts]: No need for this - it's returning anyway!

File: src/app/app.module.ts (hunk 2 above)
Comment (chungvl) [reply src/app/app.module.ts]: Just tried that:
`Error: Uncaught (in promise): Error: No component factory found for ObservationPage. Did you add it to @NgModule.entryComponents?
Error: No component factory found for ObservationPage. Did you add it to @NgModule.entryComponents?`

File: src/pages/home/home.ts (hunk 4 above)
Comment (chungvl) [reply src/pages/home/home.ts]: Fixed

File: src/pages/home/home.ts (hunk 3 above)
Comment (chungvl) [reply src/pages/home/home.ts]: Fixed

File: src/app/app.module.ts (hunk 2 above)
Comment (chungvl) [reply src/app/app.module.ts]: Fixed

File: src/app/app.component.ts (hunk 1 above)
Comment (chungvl) [reply src/app/app.component.ts]: I was getting `TS2559: Type 'string' has no properties in common with type 'Dataset'`

**Pull request merged**